src/foam/nanos/crunch/PredicatedPrerequisiteCapabilityJunctionDAO.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch',
+  name: 'PredicatedPrerequisiteCapabilityJunctionDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  documentation: `
+    A DAO decorator which filters prerequisitejunctions returned by find/select by the predicate
+    set on the junction.
+  `,
+
+  javaImports: [
+    'foam.core.X',
+    'foam.mlang.predicate.AbstractPredicate',
+    'foam.mlang.predicate.Predicate',
+    'foam.nanos.auth.Subject',
+    'foam.nanos.auth.User',
+    'static foam.mlang.MLang.*'
+  ],
+
+  methods: [
+    {
+      name: 'isAdmin',
+      args: [
+        { name: 'x', javaType: 'foam.core.X' }
+      ],
+      type: 'Boolean',
+      javaCode: `
+        Subject subject = (Subject) x.get("subject");
+        if ( subject != null ) {
+          User user = subject.getUser();
+          if ( user != null ) {
+            return user.isAdmin();
+          }
+        }
+        return false;
+      `
+    },
```

Comment:
It'd be better to use a permission specific to the logic we're permitting like predicatedprequisite.read.*
Admins will still have access since they have a star permission and we'd be able to permit other users via the permission system if we wanted to.

Reply:
addressed in : 
https://github.com/foam-framework/foam2/pull/4950/commits/55f369bfd58c26b74a8681397a7bacef56858561
https://github.com/foam-framework/foam2/pull/4950/commits/05bb0e9b912801b4df228f532f497b999e14aeca
https://github.com/foam-framework/foam2/pull/4950/commits/14db502507d7240dce4386caee92410032020976